packages/blocks/src/api/registration.js

```diff
@@ -723,6 +723,10 @@ export const getBlockVariations = ( blockName, scope ) => {
  * ```
  */
 export const registerBlockVariation = ( blockName, variation ) => {
+	if ( typeof variation.name !== 'string' ) {
+		console.warn( 'Variation names must be unique strings.' );
```

Comment:
We currently use `console.error` for similar errors during block (un)registration with `registerBlockType` and `unregisterBlockType`. Is there a good reason to prefer a warning to an error? Perhaps because we want the application to continue running as usual (which is why we don't return early too)?

Reply:
I decided to use the warning because, unlike `registerBlockType`, this doesn't prevent registration.

It should be more of a hit to folks that they're incorrectly registering block variation instead of a breaking change.